frontend/javascripts/oxalis/view/right-border-tabs/segments_tab/segments_view.tsx

```diff
@@ -1842,6 +1885,55 @@ class SegmentsView extends React.Component<Props, State> {
     );
   }
 
+  getExpandSubgroupsItem(groupId: number) {
+    const children = this.getSubGroupsAsTreeNodes(groupId);
+    const areAllChildrenExpanded =
```

Comment:
there are several spots where you use `include()` which has a linear performance overhead. combined with calling it in a collection, this becomes O(n²). initially, I thought, `expandedGroupKeys` would be small usually, but I think it can become quite large if users prefer keeping their groups open. to circumvent this, I'd define a `const expandedKeySet = new Set(this.state.expandedGroupKeys)` and then you can do a look up via `expandedKeySet.has(key)`.

Reply:
nice, I think I can actually tell that its faster now!